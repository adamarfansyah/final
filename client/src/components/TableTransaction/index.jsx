import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { renderDate } from '@utils/generateTimeSchedule';
import formattedNumber from '@utils/formattedNumber';
import Button from '@components/Button';
import classes from './style.module.scss';

const TableTransaction = ({ transactions, navigate }) => (
  <div className={classes.tableContainer} data-testid="TableTransaction">
    <table className={classes.table}>
      <thead>
        <tr>
          <th>
            <FormattedMessage id="table_venue_name" />
          </th>
          <th>
            <FormattedMessage id="table_order_id" />
          </th>
          <th>
            <FormattedMessage id="table_date" />
          </th>
          <th>
            <FormattedMessage id="table_amount" />
          </th>
          <th>
            <FormattedMessage id="table_detail" />
          </th>
        </tr>
      </thead>
      <tbody>
        {transactions.map((transaction, index) => (
          <tr key={index}>
            <td className={classes.venueName}>{transaction.BookedVenue.name}</td>
            <td>{transaction.orderId}</td>
            <td>{renderDate(transaction.startBook)}</td>
            <td data-testid="transaction-amount">{formattedNumber(transaction.amount)}</td>
            <td>
              <Button onClick={() => navigate(transaction.id)}>
                <FormattedMessage id="table_detail" />
              </Button>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

TableTransaction.propTypes = {
  transactions: PropTypes.array,
  navigate: PropTypes.func,
};

export default TableTransaction;
