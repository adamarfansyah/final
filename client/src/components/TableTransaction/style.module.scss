@import '@styles/base';
.tableContainer {
  overflow-x: auto;
}
.table {
  width: 100%;
  border-collapse: collapse;
  margin-top: 20px;

  th,
  td {
    text-align: center;
  }

  th {
    background-color: transparent;
    border-bottom: 2px solid var(--color-button);
  }
  td {
    padding: 0.5rem;
    font-size: 0.8rem;
    color: var(--color-text);
  }
  .venueName {
    font-weight: bold;
    color: $color-green-md;
  }
}
