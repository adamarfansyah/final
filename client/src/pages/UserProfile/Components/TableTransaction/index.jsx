import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

import { renderDate } from '@utils/generateTimeSchedule';
import formattedNumber from '@utils/formattedNumber';
import Button from '@components/Button';
import classes from './style.module.scss';

const TableTransaction = ({ transactions }) => {
  const navigate = useNavigate();

  const goToDetailTransaction = (transactionId) => {
    navigate(`/transaction/${transactionId}`);
  };

  return (
    <div className={classes.tableContainer}>
      <table className={classes.table}>
        <thead>
          <tr>
            <th>Venue Name</th>
            <th>Order ID</th>
            <th>Date</th>
            <th>Amount</th>
            <th>Detail</th>
          </tr>
        </thead>
        <tbody>
          {transactions.map((transaction, index) => (
            <tr key={index}>
              <td className={classes.venueName}>{transaction.BookedVenue.name}</td>
              <td>{transaction.orderId}</td>
              <td>{renderDate(transaction.startBook)}</td>
              <td>{formattedNumber(transaction.amount)}</td>
              <td>
                <Button onClick={() => goToDetailTransaction(transaction.id)}>Detail</Button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

TableTransaction.propTypes = {
  transactions: PropTypes.object,
};

export default TableTransaction;
