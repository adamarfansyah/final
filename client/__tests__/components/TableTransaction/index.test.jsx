import { render, fireEvent } from '@utils/testHelper';
import TableTransaction from '@components/TableTransaction';

const mockNavigate = jest.fn();

const mockProps = [
  {
    id: 1,
    BookedVenue: {
      name: 'Lapangan gbk 1',
    },
    orderId: 'Lapangan GBK 1-1702866864-2',
    startBook: '2023-12-17T07:00',
    endBook: '2023-12-17T08:00',
    amount: '20000000',
  },
  {
    id: 2,
    BookedVenue: {
      name: 'Lapangan gbk 2',
    },
    orderId: 'Lapangan GBK 2-1402866864-2',
    startBook: '2023-12-17T07:00',
    endBook: '2023-12-17T08:00',
    amount: '20000000',
  },
];

let wrapper;

beforeEach(() => {
  wrapper = render(<TableTransaction transactions={mockProps} navigate={mockNavigate} />);
});

describe('Table Transaction', () => {
  test('Should render correctly', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('TableTransaction')).toBeInTheDocument();
  });

  test('Should get transaction id when button clicked', () => {
    const { queryAllByTestId } = wrapper;
    const buttons = queryAllByTestId('btn');
    const button = buttons[0];

    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith(mockProps[0].id);
  });

  test('Should display correct formatted amount', () => {
    const { queryAllByTestId } = wrapper;
    const amounts = queryAllByTestId('transaction-amount');
    const amount = amounts[0];
    expect(amount).toBeInTheDocument();
  });

  test('Should match with snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});
