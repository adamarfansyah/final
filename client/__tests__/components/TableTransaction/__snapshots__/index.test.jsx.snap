// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Transaction Should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="tableContainer"
    data-testid="TableTransaction"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Nama Lapangan
          </th>
          <th>
            ID Pesanan
          </th>
          <th>
            Tanggal
          </th>
          <th>
            Jumlah
          </th>
          <th>
            Detail
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="venueName"
          >
            Lapangan gbk 1
          </td>
          <td>
            Lapangan GBK 1-1702866864-2
          </td>
          <td>
            17 December 2023
          </td>
          <td
            data-testid="transaction-amount"
          >
            Rp 20.000.000,00
          </td>
          <td>
            <button
              class="btn"
              data-testid="btn"
              type="submit"
            >
              Detail
            </button>
          </td>
        </tr>
        <tr>
          <td
            class="venueName"
          >
            Lapangan gbk 2
          </td>
          <td>
            Lapangan GBK 2-1402866864-2
          </td>
          <td>
            17 December 2023
          </td>
          <td
            data-testid="transaction-amount"
          >
            Rp 20.000.000,00
          </td>
          <td>
            <button
              class="btn"
              data-testid="btn"
              type="submit"
            >
              Detail
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
